### ISSUE TITLE
[PRE REVIEW]: hIPPYlib-MUQ: Scalable Markov Chain Monte Carlo Sampling Methods for Large-scale Bayesian Inverse Problems Governed by PDEs

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2850]
**Submitting author:** @ktkimcode (<a href="http://orcid.org/0000-0002-5929-4061">Ki-Tae Kim</a>)
**Repository:** <a href="https://github.com/hippylib/hippylib2muq.git" target ="_blank">https://github.com/hippylib/hippylib2muq.git</a>
**Version:** v0.1.0
**Editor:** Pending
**Reviewer:** Pending
**Managing EiC:** Daniel S. Katz

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**Author instructions**

Thanks for submitting your paper to JOSS @ktkimcode. **Currently, there isn't an JOSS editor assigned** to your paper.

@ktkimcode if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission (please start at the bottom of the list).

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Failed to discover a `Statement of need` section in paper

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.84  T=0.18 s (202.6 files/s, 26248.4 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Markdown                         6            371              0           1038
Python                          14            375            601            681
Jupyter Notebook                 1              0            901            257
TeX                              1             19              0            169
reStructuredText                11             44             65             63
Dockerfile                       1             10              3             27
YAML                             1              1              2             18
make                             1              4              7              9
-------------------------------------------------------------------------------
SUM:                            36            824           1579           2262
-------------------------------------------------------------------------------


Statistical information for the repository '0aab0c0373e8f352b268bc10' was
gathered on 2020/11/20.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Ki-Tae Kim                      16          2100            498           96.90
Umberto Villa                   11            69             14            3.10

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Ki-Tae Kim                 1595           76.0          0.5               24.33
Umberto Villa                62           89.9          0.5               46.77
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02850/joss.02850/10.21105.joss.02850.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02850/joss.02850/10.21105.joss.02850.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/joss.00940 is OK
- 10.1137/12089586X is OK
- 10.1137/130934805 is OK
- 10.1007/978-3-319-52462-7 is OK
- 10.1007/978-3-642-23099-8 is OK
- 10.1145/1731022.1731030 is OK
- 10.1080/10618600.1998.10474787 is OK
- 10.1007/978-3-319-11259-6_23-1 is OK
- 10.1007/b138659 is OK

MISSING DOIs

- 10.1137/17m1134640 may be a valid DOI for title: Transport map accelerated Markov chain Monte Carlo
- 10.1137/120890715 may be a valid DOI for title: Adaptive Smolyak pseudospectral approximations
- 10.1109/mlsp.2010.5589113 may be a valid DOI for title: Kalman filtering and smoothing solutions to temporal Gaussian process regression models

INVALID DOIs

- https://doi.org/10.1016/j.jcp.2015.10.008 is INVALID because of 'https://doi.org/' prefix
```

**Comment by danielskatz** [comment]
@ktkimcode - due to the small size of this code, the editors will now discuss if it meets the [substantial scholarly effort](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort) criterion for review by JOSS.  We should get back to you sometime next week.  If you want to fix the DOIs (noting that whedon's suggestions are not always right), you can, then use the following commands (one at a time, as the first line of a new comment) to regenerate the PDF and check the references.
* ```@whedon generate pdf```
* ```@whedon check references```

**Comment by danielskatz** [comment]
@whedon query scope

**Comment by whedon** [comment]
Submission flagged for editorial review.

**Comment by ktkimcode** [comment]
@danielskatz - Thanks very much for giving us a prompt feedback regarding our submission. We understand your concerns regarding the relatively small size of the project and its young age. Please allow us to provide additional information that can help you and the editorial board to assess the substantial scholarly effort criteria for consideration in JOSS.

- **Age of software**: The design and implementation of hippylib2muq is the results of a one-year long collaboration between hIPPYlib and MUQ developers led by @ktkimcode at UC Merced.

- **Number of commits**: Because the development of hippylib2muq mostly happened in a private repository and only the final product was published on GitHub, number of commits may not a reliable indicator of scholarly effort for this contribution.

- **Number of authors**: hippylib2muq is the fruits of several months of discussions between MUQ and hIPPYlib developers to design parsimonious interface between the two softwares. @ktkimcode has been leading such discussions, as well as taking care of the implementation

- **Lines of Code**: Although hippylib2muq itself is only 2,262 LOC, substantial work was done within hIPPYlib and MUQ by the co-developers of these libraries (co-authors on this paper) to make this interface possible. In particular,  hippylib2muq led to substantial development in MUQ to provide the necessary pybind wrappers to many advanced MUQ features.

- **Whether the software has already been cited in academic papers**: Not applicable since the software is brand new.

- **Whether the software is sufficiently useful that it is likely to be cited by your peer group**. As discussed in the paper, hIPPYlib-MUQ integrates two powerful open-source projects to provide unique capabilities for solving Bayesian inference problems governed by PDE forward problems. We believe that the computational & statistical inverse problems communities will greatly benefit from our software.
 
We thank the editor for his time and look forward hearing from the editors about the best way to proceed.
@ktkimcode, @uvilla, @mparno, @npetra

PS: We fixed the DOI issues.

**Comment by arfon** [comment]
@whedon reject

@ktkimcode - many thanks for your submission to JOSS. After consulting with the JOSS editorial team, we've determined that this doesn't meet our threshold for [substantial scholarly effort](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort). 

One possibility we discussed would be for this work to be contributed back to the [`hIPPYlib`](https://github.com/hippylib/hippylib) package and for there to be a follow-up (second) paper on `hIPPYlib` sometime in the future. 

JOSS has in the past published multiple papers about the same package, provided that any new major release is determined to exceed the [substantial scholarly effort](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort) threshold.
